src/hotspot/share/opto/superword.cpp

```diff
@@ -1996,291 +1961,20 @@ void PackSet::verify() const {
-            n->dump();
-          }
-        }
-      }
-#endif
-    }
-  }
-};
-
-// The C2 graph (specifically the memory graph), needs to be re-ordered.
-// (1) Build the PacksetGraph. It combines the dependency graph with the
-//     packset. The PacksetGraph gives us the dependencies that must be
-//     respected after scheduling.
-// (2) Schedule the PacksetGraph to the memops_schedule, which represents
-//     a linear order of all memops in the body. The order respects the
-//     dependencies of the PacksetGraph.
-// (3) If the PacksetGraph has cycles, we cannot schedule. Abort.
-// (4) Use the memops_schedule to re-order the memops in all slices.
-void SuperWord::schedule() {
-  if (_packset.length() == 0) {
-    return; // empty packset
-  }
-  ResourceMark rm;
-
-  // (1) Build the PacksetGraph.
-  PacksetGraph graph(this);
-  graph.build();
-
-  // (2) Schedule the PacksetGraph.
-  Node_List memops_schedule = graph.schedule();
-
-  // (3) Check if the PacksetGraph schedule succeeded (had no cycles).
-  // We now know that we only have independent packs, see verify_packs.
-  // This is a necessary but not a sufficient condition for an acyclic
-  // graph (DAG) after scheduling. Thus, we must check if the packs have
-  // introduced a cycle. The SuperWord paper mentions the need for this
-  // in "3.7 Scheduling".
-  if (!graph.schedule_success()) {
+// We prepare the memory graph for the replacement of scalar memops with vector memops.
+// We reorder all slices in parallel, ensuring that the memops inside each slice are
+// ordered according to the _schedule. This means that all packed memops are consecutive
+// in the memory graph after the reordering.
+void VTransformGraph::apply_memops_reordering_with_schedule() const {
 #ifndef PRODUCT
-    if (is_trace_superword_rejections()) {
-      tty->print_cr("SuperWord::schedule found cycle in PacksetGraph:");
-      graph.print(true, false);
-      tty->print_cr("removing all packs from packset.");
-    }
-#endif
-    _packset.clear();
-    return;
-  }
-
-#ifndef PRODUCT
-  if (is_trace_superword_info()) {
-    tty->print_cr("SuperWord::schedule: memops_schedule:");
-    memops_schedule.dump();
+  if (_is_trace_info) {
+    print_memops_schedule();
   }
 #endif
 
-  CountedLoopNode* cl = lpt()->_head->as_CountedLoop();
-  phase()->C->print_method(PHASE_SUPERWORD1_BEFORE_SCHEDULE, 4, cl);
-
-  // (4) Use the memops_schedule to re-order the memops in all slices.
-  schedule_reorder_memops(memops_schedule);
-}
-
-
-// Reorder the memory graph for all slices in parallel. We walk over the schedule once,
-// and track the current memory state of each slice.
-void SuperWord::schedule_reorder_memops(Node_List &memops_schedule) {
+  ResourceMark rm;
   int max_slices = phase()->C->num_alias_types();
-  // When iterating over the memops_schedule, we keep track of the current memory state,
+  // When iterating over the schedule, we keep track of the current memory state,
```

Comment:
Note: GitHub may present this a little badly, but I'm just moving `SuperWord::schedule_reorder_memops` -> `VTransformGraph::apply_memops_reordering_with_schedule`. This means that we now only reorder the memory graph if we actually have committed to applying the vectorization.